npm/vite-dev-server/src/plugins/cypress.ts

```diff
@@ -38,6 +38,8 @@ export const Cypress = (
   const indexHtmlFile = options.cypressConfig.indexHtmlFile
 
   let specsPathsSet = getSpecsPathsSet(specs)
+  // TODO: use async fs methods here
+  // eslint-disable-next-line no-restricted-syntax
   let loader = fs.readFileSync(INIT_FILEPATH, 'utf8')
```

Comment:
Does this "do not use sync FS calls" also need to be applied in npm, or just things we know are shipping in the binary (and never published separately on npm). It's possible a user is consuming these outside of Cypress (however unlikely) and they may not have access to wrapped `fs` module with async extensions. 

Seems safest to either us things we know exist in the Node.js standard lib, or have these depend on `fs-extra`, or use `fs.promises` (didn't know about this!) to making async versions of the functions we need in each `npm` module.

Reply:
I'd use `fs.promises` here if I was gonna change this. I started to, but then stopped because it actually requires a user-facing (I think?) type change to make this function async.

Since this is most likely to run in the plugins process, it should be fine to just use `fs.promises`, since we already run `gracefulFs.gracefulify(fs)` in the child process:

Reply:
Not too sure this would require a user facing change (this function is internal to the Vite dev server) but sure, happy to leave it like this for now. If we do change it `fs.promises` seems good.